From 4f157684f1a91520d4fbdc33504425322cf36333 Mon Sep 17 00:00:00 2001
From: Dan Villiom Podlaski Christiansen <dchristiansen@zimbra.com>
Date: Tue, 20 Jan 2015 15:11:51 +0100
Subject: [PATCH] Accessible labels for form comboboxes

The label for attribute values for the border color and background
color edit fields (Table Properties - Advanced tab) did not match the
corresponding id attribute values of the input fields.

This change ensures the for attribute value matches the id attribute
of the INPUT field, but adding explicit support for the trick used by
comboboxes for pointing to their INPUT element.
---
diff -aur -x node_modules /tmp/root/tinymce-4.9.11/src/ui/main/ts/Form.ts tinymce-4.9.11/src/ui/main/ts/Form.ts
--- /tmp/root/tinymce-4.9.11/src/ui/main/ts/Form.ts	2020-07-13 04:14:19.000000000 +0000
+++ tinymce-4.9.11/src/ui/main/ts/Form.ts	2022-08-10 05:58:01.755201122 +0000
@@ -62,17 +62,25 @@
 
     // Wrap any labeled items in FormItems
     items.each(function (ctrl) {
-      let formItem;
+      let formItem, inputId;
       const label = ctrl.settings.label;
 
       if (label) {
+        inputId = ctrl._id;
+
+        // point to the INPUTs of comboxes
+        // see the corresbonding TODO in ComboBox.ts
+        if (ctrl.subinput) {
+          inputId += '-' + ctrl.ariaTarget;
+        }
+
         formItem = new FormItem(Tools.extend({
           items: {
             type: 'label',
             id: ctrl._id + '-l',
             text: label,
             flex: 0,
-            forId: ctrl._id,
+            forId: inputId,
             disabled: ctrl.disabled()
           }
         }, self.settings.formItemDefaults));
